server/lib/pbench/indexer.py

```diff
@@ -1195,112 +1252,108 @@ def gen_sources(results, sublists, iteration, cvt_ts):
+                            if not tseries:
+                                obj.counters['sample_empty_timeseries'] += 1
                                 continue
-                            else:
-                                if not tseries:
-                                    self.counters['sample_empty_timeseries'] += 1
-                                    continue
-                            start = tseries[0]
-                            end = tseries[-1]
+                        start = tseries[0]
+                        end = tseries[-1]
+                        try:
+                            start_ts = cvt_ts(start['date'])
+                            end_ts = cvt_ts(end['date'])
+                        except KeyError:
+                            obj.counters['timeseries_missing_date'] += 1
+                            continue
+                        except BadDate:
+                            # Ignore entire sample if start/end timestamps
+                            # are bad.  Already counted.
+                            continue
+                        del sample['timeseries']
+                        # Remember the order in which samples were
+                        # processed (zero based).
+                        sample_md['@idx'] = sample_idx
+                        sample_idx += 1
+                        # Construct the same name (one based)
+                        sample_md['name'] = "sample{:d}".format(sample_idx)
+                        sample_md_subset = _dict_const([
+                                ( 'name', sample_md['name'] ),
+                                ( '@idx', sample_md['@idx'] ),
+                                ( 'uid', sample_md['uid'] ),
+                                ( 'measurement_type', sample_md['measurement_type'] ),
+                                ( 'measurement_idx', sample_md['measurement_idx'] ),
+                                ( 'measurement_title', sample_md['measurement_title'] )
+                            ])
+                        sample_md['start'] = start_ts
+                        sample_md['end'] = end_ts
+                        # Now we can emit the sample document knowing the
+                        # ID of its iteration parent document.
+                        source = _dict_const([
+                            ( '@timestamp', start_ts ),
+                            ( '@timestamp_original', start['date'] ),
+                            # Sample documents inherit the run and
+                            # iteration data of its parent.
+                            ( 'run', iteration['run'] ),
+                            ( 'iteration', iteration['iteration'] ),
+                            ( 'benchmark', iteration['benchmark'] ),
+                            ( 'sample', sample_md )
+                        ])
+                        sample_id = PbenchData.make_source_id(source)
+                        yield source, None, 'sample'
```

Comment:
I don't understand this: why None? where is the ID of the parent iteration doc?

Reply:
This is the same code as before.  Are the comments at line 1310 and 1311 sufficient?

Basically, the "`source`"  document we are yielding here *is* the parent document used for all the samples that are generated below, see line 1356, where "`sample_id`" is referenced.

Reply:
OK - thanks: I get it now - I think the comment is clear enough, but it didn't penetrate the dense fog in my skull at the time I read it.